graphicsmagick/src/wand/magick.rs

```diff
@@ -28,6 +29,44 @@ use crate::types::OrientationType;
 #[cfg(feature = "v1_3_22")]
 use crate::types::GravityType;
 
+/// # Safety
+///
+/// `STORAGE_TYPE` must match `TARGET`.
+pub unsafe trait MagickWandExportTypeSealed {
+    const STORAGE_TYPE: StorageType;
+
+    /// Make using it in generic code easier.
+    type Target: fmt::Debug + fmt::Display + Copy + Clone;
+}
+
+pub trait MagickWandExportType: MagickWandExportTypeSealed {}
```

Comment:
Can you tell me the intention here? Why is it divided into two traits.

Reply:
`MagickWandExportTypeSealed` is made private so that user cannot implement it for their own type, since they could set `STORAGE_TYPE` to `c_float` but set `Target` to `c_uchar`.